src/js/readinglist.js

```diff
@@ -332,6 +351,62 @@ function filterReadingList (e) {
   })
 }
 
+/**
+ * Toggles the buttons, and updates the options for which reading list to view.
+ */
+function changeView () {
+  this.classList.add('active')
+  // // Reset search.
+  // document.getElementById('my-search').value = ''
+  const viewAll = this.id === 'all'
+  // Updates the view setting in setting menu
+  updateOptions(viewAll)
+  // Update the button on display
+  if (this.classList.contains('right-button')) {
+    document.getElementById('all').classList.remove('active')
+    document.getElementById('reading-list').classList.add('unread-only')
+  } else {
+    document.getElementById('unread').classList.remove('active')
+    document.getElementById('reading-list').classList.remove('unread-only')
+  }
+}
+
+const isFirefox = typeof InstallTrigger !== 'undefined'
+const defaultSettings = {
+  settings: {
+    theme: 'light',
+    addContextMenu: true,
+    animateItems: !isFirefox,
+    viewAll: true
+  }
+}
+
+/**
+ *  Saves viewAll option to chrome.storage
+ * @param {boolean} viewAll The boolean value to set if all items have been viewed
+ */
+function updateOptions (viewAll) {
+  chrome.storage.sync.get(defaultSettings, items => {
+    items.settings.viewAll = viewAll
+    chrome.storage.sync.set({
+      settings: items.settings
+    })
+  })
+}
+
+/**
+ * Sets the item in the reading list to viewed.
+ *
+ * @param {string} url the url of the reading item to set to true.
+ */
+function setReadingItemViewed (url) {
+  chrome.storage.sync.get(null, pages => {
```

Comment:
Why not get only the one reading item instead of the whole list?
```javascript
chrome.storage.sync.get(url, …
```
The rest should be the same.